backend/dataall/base/aws/iam.py

```diff
@@ -66,6 +67,25 @@ def get_role_policy(
             log.error(f'Failed to get policy {policy_name} of role {role_name} : {e}')
             return None
 
+    @staticmethod
+    def list_policy_names_by_policy_pattern(account_id: str, region: str, policy_filter_pattern: str):
+        try:
+            client = IAM.client(account_id, region)
+            # Setting Scope to 'Local' to fetch all the policies created in this account
+            paginator = client.get_paginator('list_policies')
+            policies = []
+            for page in paginator.paginate(Scope='Local'):
+                policies.extend(page['Policies'])
+            policy_names = [policy.get('PolicyName') for policy in policies]
+            return [policy_nm for policy_nm in policy_names if re.search(policy_filter_pattern, policy_nm)]
```

Comment:
Modified this to filter through the list of policies which match the regex patter `{base_policy_name}-\d` ( e.g. ; datall-env-<uri>-role_name-0, datall-env-<uri>-role_name-1 , ... )